### PR TITLE
Dc el delete items

### LINKED ISSUE #12: 12. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future
## Summary
Users might make a mistake when entering an item, or may decide they won’t be buying a certain item again in the future. Allowing them to delete list items will help them keep their lists tidy.
 
## Acceptance criteria
- [ ] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [ ] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [ ]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Notes:
* Activating the “Delete” button should show a confirmation dialog. We want to give users a chance to confirm “destructive” actions.
* It might be tempting to create your own modal dialog component to handle confirmation, but modals are fraught with accessibility concerns, so consider using a simple [JavaScript confirm dialog](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm).

### PR DESCRIPTION
**Pull request by eva-lng**
|   | Type |
| --- | ---|
| ✓ | New Feature |
| ✓ | Delete an item from the list |

## Description

A delete button is rendered next to each item in a list. When clicked, the item is deleted from the database.

## Related Issue

closes #12

## Acceptance Criteria

- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Type of Changes

`new feature`

## Updates

**Updated UI with rendered delete buttons and pop up window**
Screenshot 2024-09-16 at 22 06 04

## Testing Steps / QA Criteria

1. npm start
2. navigate to List view
3. click on the delete button next to one of the items
4. confirm the pop up message

### CODE REVIEW

**Review by tataw-cl — APPROVED** [review]
Excellent work done guys. It works as expected 👌🏾. And whooooo, y'all's speed was something else 😅🔥

File: src/api/firebase.js [hunk 1]
```diff
@@ -219,10 +220,17 @@ export async function updateItem(listPath, itemId, { totalPurchases }) {
...
 	 * to delete an existing item. You'll need to figure out what arguments
 	 * this function must accept!
 	 */
+	try {
+		const itemRef = doc(db, listPath, 'items', itemId);
+		await deleteDoc(itemRef);
+		return 'Item has been deleted successfully';
+	} catch (error) {
+		console.log(error);
+	}
```
Comment (Amaka202) [inline src/api/firebase.js]: you can throw the error here and catch it in the component where you are calling deleteItem for appropriate handling

File: src/views/List.jsx [hunk 2]
```diff
@@ -28,6 +28,14 @@ export function List({ data, listPath, lists }) {
 		});
 	};
 
+	const handleDelete = async (itemId) => {
+		try {
+			await deleteItem(listPath, itemId);
+		} catch (error) {
+			console.error('Failed to delete the item', error);
+		}
```
Comment (Amaka202) [inline src/views/List.jsx]: you already have a system of error management from your past issues, you can use it here to show the user an error message. it is always good for a developer to show a good error message to a user as to why an action could not be completed.

**Review by Amaka202 — CHANGES_REQUESTED** [review]
Well done team!, your feature works mostly.💥
but when I have an internet issue and my delete request could not be completed, I don't see an error message saying why. That will def reduce the users' experience on our app.

I left a suggestion on how you can go about it😊. Let me know if you have any questions!

cc: @didemydn @eva-lng

**Comment by GrannyYetta** [comment]
Well done! The item deletion works well for me :) Congrats on the speediest issue yet!

File: src/views/List.jsx [hunk 3]
```diff
@@ -28,6 +29,16 @@ export function List({ data, listPath, lists }) {
 		});
 	};
 
+	const handleDelete = async (itemId) => {
+		try {
+			await deleteItem(listPath, itemId);
+			setErrorMsg('');
+		} catch (error) {
+			console.error(error.message, error);
+			setErrorMsg('Failed to delete the item. Please try again!');
```
Comment (Amaka202) [inline src/views/List.jsx:38]: great! well done team!!😊💥

**Review by Amaka202 — APPROVED** [review]

**Pull request merged**